src/plugins/dashboard/public/application/embeddable/grid/dashboard_grid.tsx

```diff
@@ -263,34 +263,18 @@ class DashboardGridUi extends React.Component<DashboardGridProps, State> {
       }
     });
 
-    return _.map(panelsInOrder, (panel) => {
-      const expandPanel =
-        expandedPanelId !== undefined && expandedPanelId === panel.explicitInput.id;
-      const hidePanel = expandedPanelId !== undefined && expandedPanelId !== panel.explicitInput.id;
-      const classes = classNames({
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        'dshDashboardGrid__item--expanded': expandPanel,
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        'dshDashboardGrid__item--hidden': hidePanel,
-      });
-      return (
-        <div
-          style={{ zIndex: focusedPanelIndex === panel.explicitInput.id ? 2 : 'auto' }}
-          className={classes}
-          key={panel.explicitInput.id}
-          data-test-subj="dashboardPanel"
-          ref={(reactGridItem) => {
-            this.gridItems[panel.explicitInput.id] = reactGridItem;
-          }}
-        >
-          <EmbeddableChildPanel
-            embeddableId={panel.explicitInput.id}
-            container={this.props.container}
-            PanelComponent={this.props.PanelComponent}
-          />
-        </div>
-      );
-    });
+    return _.map(panelsInOrder, ({ explicitInput, type }) => (
```

Comment:
nice refactor for cleaning up the dashboard_grid component

Reply:
Unfortunately had to revert the splitting out this part of the code since I need a little bit more time. The React Ref isn't being past correctly.